packit/cli/init.py

```diff
@@ -31,20 +41,49 @@
     is_flag=True,
     help="Reset config to default if already exists.",
 )
+@click.option(
+    "--with_precommit",
+    default=False,
+    is_flag=True,
+    help="Automatically creates an empty precommit configuration file if missing.",
+)
 @pass_config
 @cover_packit_exception
-def init(
-    config,
-    path_or_url,
-    force,
-):
+def init(config, path_or_url, force, with_precommit):
     """
-    Create the initial Packit configuration in a repository
+    Create the initial Packit configuration in a repository and add
+    a pre-commit check to validate Packit configuration file
 
     See 'packit source-git init', if you want to initialize a repository
     as a source-git repo.
     """
     working_dir = path_or_url.working_dir
+
+    if not is_git_repo(working_dir):
```

Comment:
```suggestion
    if with_precommit and not is_git_repo(working_dir):
```

The error message doesn't make much sense if checking just for git repo.